FoodOrderingSystem.py

```diff
@@ -0,0 +1,700 @@
+                filter(CustOrderStatus.order_id == order_id)
+                # label('total_price'), func.sum(CustOrderSelection.food_qty*FoodDetails.price)
+                # (func.sum(CustOrderSelection.food_qty*FoodDetails.price).label("total_price"))
+            return view
+        except Exception as ex:
+            print("Error getting order, error={}".format(str(ex)))
+        finally:
+            session.close()
+
+    def view_revenue_today(self, order_status, session):
+        """ View revenue of today """
+        # Sum all the total price of the delieverd orders
+
+        try:
+            revenue_today = session.query(FoodDetails, CustOrderSelection, CustomerDetails, CustOrderStatus).\
+            filter(CustomerDetails.cust_id == CustOrderStatus.cust_id).\
+            filter(CustOrderSelection.order_id == CustOrderStatus.order_id).\
+            filter(CustOrderSelection.food_id == FoodDetails.food_id).\
+            filter(CustOrderStatus.order_status == order_status).\
+            filter(CustOrderStatus.checkout_time <= datetime.today())
+            # label('total_price'), func.sum(CustOrderSelection.food_qty*FoodDetails.price)
+            # (func.sum(CustOrderSelection.food_qty*FoodDetails.price).label("total_price"))
+            return revenue_today
+        except Exception as ex:
+            print("Error getting order, error={}".format(str(ex)))
+        finally:
+            session.close()
+
+    def view_revenue_month(self, session):
+        """ View revenue of particular month """
+
+
+        # Get all orders of a particular month
+        # Sum all those orders
+        # Use CustOrderSelection table
+        # Need to check how to get a range or month
+        pass
+
+class Customer:
+    """ Customer's end operation """    
+    def customer_signup(self, cust_name, cust_phone, cust_email, session):
+        """ Add a new customer """
+        signup = CustomerDetails(cust_name=cust_name, cust_phone=cust_phone, cust_email=cust_email)
+        session.add(signup)
+        try:
+            session.commit()
+            return signup
+        except:
+            session.rollback()
+            raise Exception("User exist!")
+        finally:
+            session.expunge_all()
+            session.close()
+
+    def customer_login(self, cust_id, session):
+        """ Customer can login into their account """
+        try:
+            login = session.query(CustomerDetails).filter_by(cust_id=cust_id).first()
+        except Exception as ex:
+            print("Error getting customer ID, error={}".format(str(ex)))
+        finally:
+            session.close()
+        return login    
+         
+    def view_menu(self, session):
+        """ Customer can view menu """
+        try:
+            menu = session.query(FoodCategory, FoodDetails).filter(FoodCategory.category_id == FoodDetails.category_id).all()
+            return menu
```

Comment:
For view_menu(), only return the columns that matter, not all columns from all tables. (imagine seeing the menu yourself.)

Reply:
view_menu() will return:

print("\nFood ID: {} \nFood category: {} \nFood name: {} \nFood price: {}".format(fd.food_id, fc.name, fd.food_name, fd.price))